Tests/SonarScanner.MSBuild.Shim.Test/ScannerEngineInputGeneratorTest.GenerateProperties.cs

```diff
@@ -278,12 +278,14 @@ private class PropertiesContext
         public readonly ScannerEngineInput EngineInput;
         public readonly ProjectData Project;
         private readonly TestRuntime runtime;
+        private readonly IAnalysisPropertyProvider provider;
 
         public PropertiesContext(TestContext testContext, string language, TestRuntime runtime)
         {
             this.runtime = runtime;
             Config = new AnalysisConfig { SonarOutputDir = testContext.TestRunDirectory };
             EngineInput = new ScannerEngineInput(Config);
+            provider = new ListPropertiesProvider();
```

Comment:
I would either pass in the one from the field class via ctor, or do not persist it at all and call `new` on the single usage side. As it's not really used for anything yet